File: sorting/sorting.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function selectionSort(){
    let arr = [20, 14, 56, 35, 46, 23];
    for(let i = 0; i < arr.length - 1; i++){
        let minIndex = i;
        for(let j = i + 1; j < arr.length; j++){
            if(arr[j] < arr[minIndex]){
                minIndex = j;
            }
        }
        let temp = arr[minIndex];
        arr[minIndex] = arr[i]
        arr[i] = temp;
    }

    console.log("Sorted array is ", arr.join(" "));

}

function bubbleSort(){
    let arr = [34, 23, 17, 10, 56, 64];
    for(let i = 1; i < arr.length - 1; i++){
        let swapped = false;
        for(let j = 0; j < arr.length - 1; j++){
            if(arr[j] > arr[j+1]){
                let temp = arr[j+1];
                arr[j+1] = arr[j];
                arr[j] = temp;
                swapped = true;
            }
        }

        if(swapped == false){
            break;
        }
    }

    console.log("Here is sorted arr ", arr.join(" "));
}

function insetionSort(){
    let arr = [34, 23, 17, 10, 56, 64];
    for(let i = 0; i < arr.length; i++){
        let temp = arr[i];
        let j = i - 1;
        for(; j >= 0; j--){
            if(arr[j] > temp){
                arr[j+1] = arr[j];
            } else {
                break;
            }
        }
        arr[j+1] = temp;
    }
    console.log(" Sorted array is here ", arr.join(" "));
}



function main() {
  console.log("\nSelect the items which you want to print from below list: \n");
  console.log();
  console.log("\n1. Selection Sort\n");
  console.log("\n2. Bubble Sort\n");

  rl.question("", (num_of_item) => {
    num_of_item = parseInt(num_of_item);
    console.log();

    switch (num_of_item) {
      case 1:
        selectionSort();
        break;
      case 2:
        bubbleSort();
        break;
      case 3:
        insetionSort();
        break;

      default:
        console.log(
          "\nInvalid Number, Please select a number from above given list :)\n"
        );
        rl.close();
        break;
    }
  });
  
  return 0;
}

main();

File: sorting/sorting.py

def selectionSort():
    arr = [23, 12, 34, 14, 56, 45, 5]
    for i in range(len(arr)-1):
        minIndex = i
        for j in range(i + 1, len(arr)):
            if(arr[j] < arr[minIndex]):
                minIndex = j
        arr[minIndex], arr[i] = arr[i], arr[minIndex]
    print(" The sorted array is ", arr)

def bubbleSort():
    arr = [23, 12, 34, 14, 56, 45, 5]
    for i in range(len(arr) - 1):
        swapped = False
        for j in range(0, len(arr) - 1):
            if(arr[j] > arr[j + 1]):
                arr[j], arr[j+1] = arr[j+1], arr[j]
                swapped = True
        if swapped == False:
            break
    print(" Here is sorted array ", arr)

def insertionSort():
    arr = [23, 12, 34, 14, 56, 45, 5]

    for i in range(len(arr)):
        temp = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > temp:
            arr[j+1] = arr[j]
            j -= 1
        arr[j + 1] = temp
    print(" Sorted array is here ", arr)

        

def invalid_item():
     print("\nInvalid Number, Please select a number from the above given list :)\n")

def main():
    print("\nSelect the items from below given list which you want to print\n")
    print("\n1. Selection Sort\n")
    print("\n2. Bubble Sort\n")
    print("\n3. Insertion sort\n")

    num_of_item = int(input("Enter the number: "))
    print()

    items = {
        1: selectionSort,
        2: bubbleSort,
        3: insertionSort,
    }

    items.get(num_of_item, invalid_item)()

if __name__ == "__main__":
    main()